### PR TITLE
[MetaCI] CTC release integration

### PR DESCRIPTION
**Pull request by jofsky**

### CODE REVIEW

File: .heroku/release.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash 
+set -e
+
+if [ -n "$CTC_URL" ] ; then
+     # Run migration surrounded by CTC start/stop
+     python .heroku/check_change_traffic_control.py
+else
+     # Simply run the migration
+     python manage.py migrate --noinput
+     python .heroku/check_change_traffic_control.py
```
Comment (davisagli) [inline .heroku/release.sh:10]: The point of the if/else is that we don't want to call check_change_traffic_control.py if there's no CTC_URL

File: .heroku/check_change_traffic_control.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    source_url = f"{settings.ctc_repo_url}/commit/{settings.heroku_slug_commit}"
+    logger.info("Checking change traffic control for {source_url}")
+
+    # find Case and start implementation step
+    result = call_ctc(
+        f"{settings.ctc_url}/case/match-and-start",
+        params={
+            "sm_business_name_id": settings.ctc_sm_business_name_id,
+            "source_url": source_url,
+            "heroku_app_name": settings.heroku_app_name,
```
Comment (davisagli) [inline .heroku/check_change_traffic_control.py:58]: step_description

**Review by davisagli — CHANGES_REQUESTED** [review]

File: .heroku/release.sh (hunk 1 above)
Comment (jofsky) [reply .heroku/release.sh:10]: Sorry I did this for my local testing.

File: .heroku/check_change_traffic_control.py (hunk 2 above)
Comment (jofsky) [inline .heroku/check_change_traffic_control.py:58]:
```suggestion
            "step_description": settings.heroku_app_name,
```

File: .heroku/release.sh (hunk 1 above)
Comment (jofsky) [inline .heroku/release.sh:10]:
```suggestion
```

**Comment by davisagli** [comment]
I think this looks ready to test once we've deployed https://github.com/SFDO-Tooling/gus-bus/pull/102

File: .heroku/check_change_traffic_control.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import jwt
+import pydantic
+import requests
+
+logger = logging.getLogger()
+
+
+class CTCSettings(pydantic.BaseSettings):
+    ctc_url: str
+    ctc_auth_key: str
```
Comment (davisagli) [inline .heroku/check_change_traffic_control.py:16]: Change this to use metaci_release_webhook_url and metaci_release_webhook_auth_key so we don't have to duplicate

Comment (jofsky) [reply .heroku/check_change_traffic_control.py]: @davisagli not sure you I correctly interpreted what you wanted, let me know.

**Review by davisagli — APPROVED** [review]

**Pull request merged**